hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecordPayload.java

```diff
@@ -207,6 +195,13 @@ static Option<String> getPayloadClassNameIfPresent(Properties props) {
       payloadClassName = ConfigUtils.getStringWithAltKeys(props, PAYLOAD_CLASS_NAME);
     } else if (props.containsKey("hoodie.datasource.write.payload.class")) {
       payloadClassName = props.getProperty("hoodie.datasource.write.payload.class");
+    } else if (ConfigUtils.containsConfigProperty(props, RECORD_MERGE_MODE)
+        && ConfigUtils.getStringWithAltKeys(props, RECORD_MERGE_MODE, StringUtils.EMPTY_STRING)
```

Comment:
can you help me understand, why do we need this refactoring ?

Reply:
we really need to know if payload class is set in TableConfig or not mainly for EightToNineUpgradeHandler right. 
we may need it for other callers. can we ensure we try not to change other callers unintentionally.

Reply:
instead, why not leverage `HoodieTableConig.propsMap()` within EightToNineUpgradeHandler to check if payload class is set or not within HoodieTableConfig  and we don't need any of these changes.